OpenEdXMobile/src/main/java/org/edx/mobile/util/ConfigUtil.kt

```diff
@@ -0,0 +1,84 @@
+ */
+object ConfigUtil {
+
+    /**
+     * Returns true if domain of the given URL is white-listed in the configuration,
+     * false otherwise.
+     *
+     * @param url
+     * @return
+     */
+    fun isWhiteListedURL(url: String, config: Config): Boolean {
+        // check if this URL is a white-listed URL, anything outside the white-list is EXTERNAL LINK
+        for (domain in config.zeroRatingConfig.whiteListedDomains) {
+            if (BrowserUtil.isUrlOfHost(url, domain)) {
+                // this is white-listed URL
+                return true
+            }
+        }
+        return false
+    }
+
+    /**
+     * Utility method to check the current build is white listed in Firebase remote config.
+     *
+     * @param config   [Config]
+     * @param listener [OnCourseUpgradeStateListener] callback for the white listed release
+     */
+    fun isReleaseWhiteListed(config: Config,
+                             listener: OnCourseUpgradeStateListener?) {
+        if (config.firebaseConfig.isEnabled) {
+            val firebaseRemoteConfig = FirebaseRemoteConfig.getInstance()
+            firebaseRemoteConfig.fetchAndActivate().addOnCompleteListener {
+                val courseUpgradeEnabled = firebaseRemoteConfig
+                        .getBoolean(AppConstants.FirebaseConstants.REV_934_ENABLED)
+                var releaseWhiteListed = false
+                if (courseUpgradeEnabled) {
+                    val whiteListedReleasesJson = firebaseRemoteConfig
+                            .getString(AppConstants.FirebaseConstants.REV_934_WHITELISTED_RELEASES)
+                    if (!TextUtils.isEmpty(whiteListedReleasesJson)) {
+                        val whiteListReleases = Gson().fromJson<ArrayList<String>>(whiteListedReleasesJson,
+                                object : TypeToken<ArrayList<String>>() {}.type)
+                        for (WhiteListedRelease in whiteListReleases) {
+                            if (BuildConfig.VERSION_NAME.equals(WhiteListedRelease, ignoreCase = true)) {
+                                listener?.onWhiteListedRelease()
+                                releaseWhiteListed = true
+                                break
+                            }
+                        }
+                    }
+                }
+                if (!releaseWhiteListed) {
+                    listener?.onCourseUpgradeDisable()
+                }
+            }
+        }
+    }
+
+    /**
+     * Interface to provide the callback for the status of the course update.
+     */
+    interface OnCourseUpgradeStateListener {
+        /**
+         * Callback about to know the current release is white listed for the course upgrade feature
+         * */
+        fun onWhiteListedRelease()
+
+        /**
+         * Callback to know that the course update is enabled or current release is block for the
+         * course upgrade feature
+         * */
+        fun onCourseUpgradeDisable()
+    }
```

Comment:
As per discussion, following code is more optimized.

```kotlin
    /**
     * Utility method to check the status of the course upgrade.
     *
     * @param config   [Config]
     * @param listener [OnCourseUpgradeStatusListener] callback for the status of the course upgrade.
     */
    fun checkCourseUpgradeEnabled(config: Config,
                                  listener: OnCourseUpgradeStatusListener?) {
        // Check firebase enabled in config
        if (config.firebaseConfig.isEnabled) {
            val firebaseRemoteConfig = FirebaseRemoteConfig.getInstance()
            firebaseRemoteConfig.fetchAndActivate().addOnCompleteListener {
                val courseUpgradeEnabled = firebaseRemoteConfig
                        .getBoolean(AppConstants.FirebaseConstants.REV_934_ENABLED)
                // Check course upgrade enabled in firebase remote config
                if (courseUpgradeEnabled) {
                    val whiteListedReleasesJson = firebaseRemoteConfig
                            .getString(AppConstants.FirebaseConstants.REV_934_WHITELISTED_RELEASES)
                    if (!TextUtils.isEmpty(whiteListedReleasesJson)) {
                        val whiteListReleases = Gson().fromJson<ArrayList<String>>(whiteListedReleasesJson,
                                object : TypeToken<ArrayList<String>>() {}.type)
                        // Check current release is white listed in firebase remote config
                        for (WhiteListedRelease in whiteListReleases) {
                            if (BuildConfig.VERSION_NAME.equals(WhiteListedRelease, ignoreCase = true)) {
                                listener?.onCourseUpgradeResult(true)
                                return@addOnCompleteListener
                            }
                        }
                    }
                }
                listener?.onCourseUpgradeResult(false)
            }
        }
    }

    /**
     * Interface to provide the callback for the status of the course upgrade.
     */
    interface OnCourseUpgradeStatusListener {
        /**
         * Callback to send course upgrade status result.
         */
        fun onCourseUpgradeResult(enabled: Boolean)
    }
```

Reply:
done.